clients/src/main/java/org/apache/kafka/common/requests/DeleteTopicsRequest.java

```diff
@@ -62,85 +59,62 @@
+        public Builder(DeleteTopicsRequestData data) {
             super(ApiKeys.DELETE_TOPICS);
-            this.topics = topics;
-            this.timeout = timeout;
+            this.data = data;
         }
 
         @Override
         public DeleteTopicsRequest build(short version) {
-            return new DeleteTopicsRequest(topics, timeout, version);
+            return new DeleteTopicsRequest(data, version);
         }
 
         @Override
         public String toString() {
-            StringBuilder bld = new StringBuilder();
-            bld.append("(type=DeleteTopicsRequest").
-                append(", topics=(").append(Utils.join(topics, ", ")).append(")").
-                append(", timeout=").append(timeout).
-                append(")");
-            return bld.toString();
+            return data.toString();
         }
     }
 
-    private DeleteTopicsRequest(Set<String> topics, Integer timeout, short version) {
+    private DeleteTopicsRequest(DeleteTopicsRequestData data, short version) {
         super(ApiKeys.DELETE_TOPICS, version);
-        this.topics = topics;
-        this.timeout = timeout;
+        this.data = data;
+        this.version = version;
     }
 
     public DeleteTopicsRequest(Struct struct, short version) {
         super(ApiKeys.DELETE_TOPICS, version);
-        Object[] topicsArray = struct.getArray(TOPICS_KEY_NAME);
-        Set<String> topics = new HashSet<>(topicsArray.length);
-        for (Object topic : topicsArray)
-            topics.add((String) topic);
-
-        this.topics = topics;
-        this.timeout = struct.getInt(TIMEOUT_KEY_NAME);
+        this.data = new DeleteTopicsRequestData(struct, version);
+        this.version = version;
     }
 
     @Override
     protected Struct toStruct() {
-        Struct struct = new Struct(ApiKeys.DELETE_TOPICS.requestSchema(version()));
-        struct.set(TOPICS_KEY_NAME, topics.toArray());
-        struct.set(TIMEOUT_KEY_NAME, timeout);
-        return struct;
+        return data.toStruct(version);
+    }
+
+    public DeleteTopicsRequestData data() {
+        return data;
     }
 
     @Override
     public AbstractResponse getErrorResponse(int throttleTimeMs, Throwable e) {
-        Map<String, Errors> topicErrors = new HashMap<>();
-        for (String topic : topics)
-            topicErrors.put(topic, Errors.forException(e));
-
-        switch (version()) {
-            case 0:
-                return new DeleteTopicsResponse(topicErrors);
-            case 1:
-            case 2:
-            case 3:
-                return new DeleteTopicsResponse(throttleTimeMs, topicErrors);
-            default:
-                throw new IllegalArgumentException(String.format("Version %d is not valid. Valid versions for %s are 0 to %d",
-                    version(), this.getClass().getSimpleName(), ApiKeys.DELETE_TOPICS.latestVersion()));
+        DeleteTopicsResponseData response = new DeleteTopicsResponseData();
+        if (version >= 3) {
```

Comment:
Should be "if (version >= 1) {" to match the previous code, right?

Reply:
Good catch! I've just pushed a fix